### PR TITLE
@sweir27 => Propagate original user agent, and append with Metaphysics

### PR DESCRIPTION
**Pull request by mzikherman**
This changes that last arg to the loaders/factory to be an object, including the request id and user agent, vs having it just be the user agent. This way, (hopefully!), if we want to add another option, we don't have to keep modifying the signature all the way around.

Closes https://github.com/artsy/metaphysics/issues/937

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -85,7 +86,10 @@ async function startApp() {
         defaultTimezone = timezone
       }
 
-      const loaders = createLoaders(accessToken, userID, requestID)
+      const loaders = createLoaders(accessToken, userID, {
```
Comment (mzikherman) [inline src/index.js]: This was confusing, here its `requestID`, and below its `requestIDs`.

However, they wind up going thru the same factory (where it's `requestIDs`), not sure what the deal is?

Either way, testing some stuff locally, I was seeing request id and user agent in the outgoing requests 👍

File: src/index.js [hunk 2]
```diff
@@ -70,6 +70,7 @@ async function startApp() {
       const accessToken = req.headers["x-access-token"]
       const userID = req.headers["x-user-id"]
       const timezone = req.headers["x-timezone"]
+      const userAgent = req.headers["user-agent"]
```
Comment (sweir27) [inline src/index.js]: Will this header _always_ be set? Wondering if this needs some fallback so we're not trying to append to a `null` value later.

Comment (mzikherman) [reply src/index.js]: I can add a fallback.

**Comment by mzikherman** [comment]
Updated to make sure to set the ua to `Metaphysics` if its not present (otherwise append).

File: src/index.js (hunk 1 above)
Comment (sweir27) [reply src/index.js]: 🤔 weird.

Comment (mzikherman) [reply src/index.js]: This doesn't look like it was actually used except in convection/merged schemas, so it's possible that its wrong (and should be `requestIDs`, or whatever), or maybe this is actually right! And it's being used properly in the `mergeSchemas` methods.

**Pull request merged**